packages/core/strapi/ee/index.js

```diff
@@ -0,0 +1,218 @@
+  }
+};
+
+const verifyLicense = (license) => {
+  const [signature, base64Content] = Buffer.from(license, 'base64').toString().split('\n');
+  const stringifiedContent = Buffer.from(base64Content, 'base64').toString();
+
+  const verifier = crypto.createVerify('RSA-SHA256');
+  verifier.update(stringifiedContent);
+  verifier.end();
+
+  const verified = verifier.verify(publicKey, signature, 'base64');
+
+  return { verified, licenseInfo: verified ? JSON.parse(stringifiedContent) : null };
+};
+
+let initialized = false;
+
+const init = (licenseDir, logger) => {
+  // Can only be executed once, to prevent any abuse of the optimistic behavior
+  if (initialized) {
+    return;
+  }
+
+  initialized = true;
+  ee.logger = logger;
+
+  if (process.env.STRAPI_DISABLE_EE?.toLowerCase() === 'true') {
+    return;
+  }
+
+  const license = process.env.STRAPI_LICENSE || readLicense(licenseDir);
+
+  if (license) {
+    const { verified, licenseInfo } = verifyLicense(license);
+
+    if (verified) {
+      ee.enabled = true; // Optimistically enable EE during initialization
+      ee.licenseInfo = licenseInfo;
+    } else {
+      return disable();
+    }
+  }
+};
+
+const oneMinute = 1000 * 60;
+
+const onlineUpdate = async (db) => {
+  const transaction = await db.transaction();
+
+  try {
+    // TODO: Use the core store interface instead, it does not support transactions and "FOR UPDATE" at the moment
+    const eeInfo = await db
+      .queryBuilder(coreStoreModel.uid)
+      .where({ key: 'ee_information' })
+      .select('value')
+      .first()
+      .transacting(transaction)
+      .forUpdate()
+      .execute()
+      .then((result) => (result ? JSON.parse(result.value) : result));
+
+    const useStoredLicense = eeInfo?.lastOnlineCheck > Date.now() - oneMinute;
+    const license = useStoredLicense
+      ? eeInfo.license
+      : await fetchLicense(ee.licenseInfo.licenseKey, eeInfo?.license);
+
+    if (license) {
+      const { verified, licenseInfo } = verifyLicense(license);
+
+      if (verified) {
+        ee.licenseInfo = licenseInfo;
+      } else {
+        disable();
+      }
+    }
+
+    if (!useStoredLicense) {
+      const value = { license, lastOnlineCheck: Date.now() };
+      const query = db.queryBuilder(coreStoreModel.uid).transacting(transaction);
+
+      if (!eeInfo) {
+        query.insert({ key: 'ee_information', value: JSON.stringify(value), type: typeof value });
+      } else {
+        query.update({ value: JSON.stringify(value) }).where({ key: 'ee_information' });
+      }
+
+      await query.execute();
+    } else if (!license) {
+      disable();
+    }
+
+    await transaction.commit();
+  } catch (error) {
+    // The database can be locked at the time of writing, seems to just be a SQLite issue
+    await transaction.rollback();
+  }
+};
+
+const defaultFeatures = {
+  bronze: [],
+  silver: [],
+  gold: ['sso'],
+};
+
+const validateInfo = () => {
+  if (!ee.licenseInfo.expireAt) {
```

Comment:
I recall that it could happen when it was previously disabled and emptied due to the structure of the old `recurringCheck` function, but I reworked it, and this shouldn't be necessary anymore.